packages/react/src/util.ts

```diff
@@ -0,0 +1,27 @@
+import { PenumbraInjection, PenumbraSymbol } from '@penumbra-zone/client';
+
+export const keyOfInjection = (injection?: PenumbraInjection) =>
+  Object.entries(window[PenumbraSymbol] ?? {}).find(
+    ([keyOrigin, valueInjection]) =>
+      keyOrigin &&
+      // matching injection
+      valueInjection === injection,
+  )?.[0];
+
+export const injectionOfKey = (keyOrigin?: string) =>
+  keyOrigin ? window[PenumbraSymbol]?.[keyOrigin] : undefined;
+
+export const assertStringIsOrigin = (s?: string) => {
+  if (!s || new URL(s).origin !== s) {
+    throw new TypeError('Invalid origin');
+  }
+  return s;
+};
+
+export const assertManifestOrigin = (s?: string, injection?: PenumbraInjection) => {
```

Comment:
question: can this be moved to the `@penumra-zone/client/create`?

Reply:
there's already methods there for this, it was duplicated and now deleted